### DISCUSSION TITLE
Discord bot library

### DISCUSSION THREAD

**Discussion by franciscofpll** [socketry/async#325] [General]
Hi! 

I recently worked on a [project](https://github.com/zedbydade/ruby-map-reduce) with the Async gem and got something that worked with a map-reduce implementation. However, it is still in very early development, so I would appreciate it if someone could give me some tips about how to use Async. 

Most importantly, during that library development, I learned that with the async gem, we could finally implement the Discord bot library in Ruby, like the Nodejs one. Does anyone know if we already have some project or are interested in developing something like that? If so, please don't hesitate to reach out to me. I will start the library development in the next few weeks.

**Comment by ioquatix** [comment]
Hello. I'm not sure how the linked code is utilizing Async. Can you explain in more detail?

I don't have any particular interest in developing a discord bot library at this time, but it sounds interesting.
